### PR TITLE
SIG-scalability charter.

### PR DESCRIPTION
**Pull request by wasylkowski-a**
This is a copy of @porridge's PR #1607. @porridge will no longer be working on that PR, so I am taking it over with his consent.

SIG-scalability leads:
/cc @wojtek-t @countspongebob
Steering committee representatives:
/cc @thockin @smarterclayton @spiffxp

### CODE REVIEW

**Comment by shyamjvs** [comment]
Thanks @wasylkowski-a!

+1 to @countspongebob's comment on addressing gaps in a later PR.

**Comment by wojtek-t** [comment]
> Content of the charter is good, but per some other comments from steering we need to address some basic governance issues like SIG leads. I'd like to see us approve this as it is the substantive content and do another PR to address remaining gaps.

I'm fine with that, but let's add *explicit TODOs* to the charter to address the issues mentioned in the original PR.

**Comment by wojtek-t** [comment]
> Would you be ok with a single to-do that covers review of inclusion of the template with all of the relevant items?

I think that we need a TODO that will describe all the missing parts in the charter. If we mention all the missing things, I'm fine with that.

**Comment by wasylkowski-a** [comment]
I will do that as soon as I get some uninterrupted time; hopefully this week. Stay tuned.

File: sig-scalability/charter.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+   * Actually designing and implementing performance improvements to the system.
+
+
+### Notes about the time allocation split above:
+* the numbers are meant as a reminder that balance is necessary to make sure that we are not completely overwhelmed by debugging, you do NOT need to do per-minute accounting of your time,
+* the split is expected to change as our tools improve and flakiness decreases - we hope to devote less time to fire-fighting and more time to performance improvements in the long term.
+
+## What can we do/require from other SIGs
+
+*By regression below we mean a regression identified by the set of release-blocking scalability/performance tests (as defined by [the sig-release-x.y-blocking dashboards](https://github.com/kubernetes/test-infra/blob/master/testgrid/config/config.yaml)).*
```
Comment (hongchaodeng) [inline sig-scalability/charter.md]: > https://github.com/kubernetes/test-infra/blob/master/testgrid/config/config.yaml

This page is "not found" :)
Can you fix it?

File: sig-scalability/charter.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+   1. this can be done by reading code changes, bisecting, running A/B tests, etc,
+   2. we say a PR is identified as the cause when we are reasonably confident that it indeed caused a regression, *even if the mechanism is not 100% understood* - this is because we want to minimize the time of merge queue pause
+4. mitigate the regression; this can potentially be different things, e.g.:
+   1. reverting the PR,
+   2. switching some feature off (preferably by default, at last resort in the tests only),
+   3. downgrading Go version,
+   4. etc
+5. once the mitigation is merged to master, unpause the merge queue
+
+### Rationale
```
Comment (hongchaodeng) [inline sig-scalability/charter.md]: These are really good points.
One suggestion: put them in the first section to justify the proposal

File: sig-scalability/charter.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+* We can require a SIG to introduce a regression-catching benchmark test for a scalability-critical functionality.
+
+"Rules of engagement" for pausing merge queue:
+1. observe a scalability regression on master,
+2. pause the merge queue,
+3. identify the PR which caused the regression
+   1. this can be done by reading code changes, bisecting, running A/B tests, etc,
+   2. we say a PR is identified as the cause when we are reasonably confident that it indeed caused a regression, *even if the mechanism is not 100% understood* - this is because we want to minimize the time of merge queue pause
+4. mitigate the regression; this can potentially be different things, e.g.:
+   1. reverting the PR,
```
Comment (timothysc) [inline sig-scalability/charter.md]: I think this needs to be handled with care to ensure folks are informed of the tradeoff's associated with a revert.  https://github.com/kubernetes/kubernetes/pull/60891#issuecomment-371240524 is a good example that blindly reverting without conversing it through can be dangerous.

**Comment by spiffxp** [comment]
Same questions as in the previous PR
- what are the responsibilities of your SIG leads?
- what process do you use to change SIG leadership?
- what code do you own?

See https://github.com/kubernetes/community/blob/master/committee-steering/governance/sig-governance-requirements.md for the checklist we use to evaluate charters

**Comment by wojtek-t** [comment]
I'm going to rewrite it to more match the style SIG Architecture recommend for writing charters.

So I'm closing this one and will hopefully open a new one within the next 2 weeks.

**Comment by wojtek-t** [comment]
@wasylkowski-a - I don't have persmissions to close this one, so please close :)

**Comment by cblecker** [comment]
@wojtek-t you should be able to close with the `/close` command :)

**Comment by wojtek-t** [comment]
I've opened https://github.com/kubernetes/community/pull/2149 as a replacement.

**Pull request closed without merging**